src/main/java/org/apache/sysds/runtime/matrix/operators/AggregateUnaryOperator.java

```diff
@@ -35,7 +35,7 @@ public class AggregateUnaryOperator extends MultiThreadedOperator {
 	private static final long serialVersionUID = 6690553323120787735L;
 
 	public final AggregateOperator aggOp;
-	public final IndexFunction indexFn;
+	public IndexFunction indexFn;
```

Comment:
if we can please keep it final.

Reply:
👍  I refactored the `CountDistinctOperator` ctor to make the API more intuitive